src/components/VersionChooser/index.tsx

```diff
@@ -29,24 +45,41 @@ class VersionChooserBase extends React.Component<PublicProps> {
             </Col>
           </Row>
 
-          <Form.Row>
-            <VersionSelect
-              label={gettext('Choose an old version')}
-              listedVersions={listedVersions}
-              unlistedVersions={unlistedVersions}
-            />
+          {versionsLists ? (
+            <Form.Row>
+              <VersionSelect
+                label={gettext('Choose an old version')}
+                listedVersions={versionsLists.listed}
+                unlistedVersions={versionsLists.unlisted}
+              />
 
-            <VersionSelect
-              label={gettext('Choose a new version')}
-              listedVersions={listedVersions}
-              unlistedVersions={unlistedVersions}
-              withLeftArrow
+              <VersionSelect
+                label={gettext('Choose a new version')}
+                listedVersions={versionsLists.listed}
+                unlistedVersions={versionsLists.unlisted}
+                withLeftArrow
+              />
+            </Form.Row>
+          ) : (
+            <Loading
+              message={gettext('Retrieving all the versions of this add-on...')}
```

Comment:
I am fine with aborting the action and turning off the loading state, but then what? What do we do?

The only issue we could have is someone unplugging the ethernet cable at this point, which is something we did not handle anywhere yet (except for the user maybe). Would it make sense to file an issue for that, for all the api calls? 

The current patch is pretty big, so that might be easier as a follow-up.

Reply:
> I am fine with aborting the action and turning off the loading state, but then what? What do we do?

Then we render the error. Until then, the console will show the error. But, yeah, it's not required right now, I guess. I find it's easier on the brain to look at the Redux action stack and see an explicit loading state rather than an implicit one. 

We can deal with it later. I won't file an issue because we could possibly still infer a loading state even when we have an error handler. We can revisit the design when adding error handling.

Reply:
I just think it's a better habit to use explicit loading states. It will make everything fall into place later on.